### PR TITLE
enhancement(commands): update holiday list to show dates

### PR DESCRIPTION
**Pull request by PatrickDohn**
#### Summary

_Enumerate and describe your changes_

- Update the holiday list command to now show the dates of each holiday.

#### Checklist

_Double check list and complete these actions before merge_

- [x] PR title is in format `TYPE(scope): description`
- [x] Link PR to Issue (if any issues are related)

### CODE REVIEW

File: src/utils/slackBlockKitUtil.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 const _ = require('lodash');
 const moment = require('moment');
 const {
-  HOLIDAYS,
   SINGLE_BACKTICK,
```
Comment (PatrickDohn) [inline src/utils/slackBlockKitUtil.js:4]: No longer needed we can maybe get rid of this from constants?

File: src/utils/slackBlockKitUtil.js [hunk 2]
```diff
@@ -273,26 +272,22 @@ class SlackBlockKitUtil {
...
-      {
+  buildHolidayList(holidays) {
+    const holidayList = _.map(holidays, (holiday) => {
+      return {
+        holiday: holiday.description,
+        date: holiday.start.date,
+      };
+    });
+    const blocks = holidayList.map((holiday) => {
+      return {
```
Comment (PatrickDohn) [inline src/utils/slackBlockKitUtil.js:283]: After making the calendar request we map over the holiday response object to get the holidays and dates.

File: src/helpers/helpers.js [hunk 3]
```diff
@@ -55,7 +55,18 @@ const parsePhrases = (phrases) => {
   return parsedPhrases;
 };
 
+const parseHolidays = (holidays) => {
```
Comment (davidholyko) [inline src/helpers/helpers.js:58]: add jsDoc

https://github.com/f1v/hrbot/blob/master/src/helpers/helpers.js#L17-L22

looks like holidays in an object, what are its keys and values?

File: src/helpers/helpers.js [hunk 4]
```diff
@@ -55,7 +55,18 @@ const parsePhrases = (phrases) => {
...
+  let arr = [];
+  for (let key in holidays) {
```
Comment (davidholyko) [inline src/helpers/helpers.js:60]: leverage lodash

https://lodash.com/docs/4.17.15

```
_.keys
_.map
```

File: src/utils/slackBlockKitUtil.js [hunk 5]
```diff
@@ -272,30 +271,6 @@ class SlackBlockKitUtil {
     return blocks;
   }
```
Comment (PatrickDohn) [inline src/utils/slackBlockKitUtil.js:273]: Remove buildHolidayList and use genericBlock instead

File: src/helpers/helpers.js (hunk 3 above)
Comment (PatrickDohn) [reply src/helpers/helpers.js:65]: Fixed

File: src/helpers/helpers.js (hunk 4 above)
Comment (PatrickDohn) [reply src/helpers/helpers.js]: Fixed

File: src/utils/googleCalendarUtil.js [hunk 6]
```diff
@@ -96,7 +96,17 @@ class GoogleCalendarUtil {
       return event.summary.includes(HOLIDAY);
     });
 
-    return eventList;
+    const holidayList = _.map(eventList, (holiday) => {
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:99]: this seems like an unnecessary step

File: src/helpers/helpers.js [hunk 7]
```diff
@@ -55,7 +55,26 @@ const parsePhrases = (phrases) => {
   return parsedPhrases;
 };
 
+/**
+ * Takes in a Object with Keys and Values that represent holidays and dates
+ * and returns an array seperated by new lines.
+ * @param {Object} holidays
+ * @returns {Array}
+ */
+
```
Comment (davidholyko) [inline src/helpers/helpers.js:64]: rm this empty new line

File: src/helpers/helpers.js [hunk 8]
```diff
@@ -55,7 +55,26 @@ const parsePhrases = (phrases) => {
...
+const parseHolidays = (holidays) => {
+  let arr = [];
+  const holidayKeys = _.keys(holidays);
+  _.map(holidayKeys, (holiday) => {
+    if (holiday) {
+      arr.push(`\n${holiday}, ${holidays[holiday]}`);
+    }
+  });
+  return arr;
```
Comment (davidholyko) [inline src/helpers/helpers.js:73]: does the list happen to return the holidays in order by date ascending? 

ie. jan first, december last

**Review by davidholyko — APPROVED** [review]
LGTM

File: src/helpers/helpers.js (hunk 7 above)
Comment (PatrickDohn) [reply src/helpers/helpers.js]: Fixed

**Pull request merged**